server/src/uuid/model/entity/mod.rs

```diff
@@ -484,18 +485,81 @@ impl Entity {
+                    payload.input.meta_description.unwrap(),
+                ),
+                ("meta_title".to_string(), payload.input.meta_title.unwrap()),
+                ("title".to_string(), payload.input.title.unwrap()),
+                ("url".to_string(), payload.input.url.unwrap()),
+            ]),
+            EntityRevisionType::Article => HashMap::from([
+                ("content".to_string(), payload.input.content.unwrap()),
+                (
+                    "meta_description".to_string(),
+                    payload.input.meta_description.unwrap(),
+                ),
+                ("meta_title".to_string(), payload.input.meta_title.unwrap()),
+                ("title".to_string(), payload.input.title.unwrap()),
+            ]),
+            EntityRevisionType::Course => HashMap::from([
+                ("content".to_string(), payload.input.content.unwrap()),
+                (
+                    "meta_description".to_string(),
+                    payload.input.meta_description.unwrap(),
+                ),
+                ("title".to_string(), payload.input.title.unwrap()),
+            ]),
+            EntityRevisionType::CoursePage => HashMap::from([
+                ("content".to_string(), payload.input.content.unwrap()),
+                ("title".to_string(), payload.input.title.unwrap()),
+            ]),
+            EntityRevisionType::Event => HashMap::from([
+                ("content".to_string(), payload.input.content.unwrap()),
+                (
+                    "meta_description".to_string(),
+                    payload.input.meta_description.unwrap(),
+                ),
+                ("meta_title".to_string(), payload.input.meta_title.unwrap()),
+                ("title".to_string(), payload.input.title.unwrap()),
+            ]),
+            EntityRevisionType::Exercise => {
+                HashMap::from([("content".to_string(), payload.input.content.unwrap())])
+            }
+            EntityRevisionType::ExerciseGroup => HashMap::from([
+                ("content".to_string(), payload.input.content.unwrap()),
+                (
+                    "cohesive".to_string(),
+                    payload.input.meta_description.unwrap(),
+                ),
+            ]),
+            EntityRevisionType::GroupedExercise => {
+                HashMap::from([("content".to_string(), payload.input.content.unwrap())])
+            }
+            EntityRevisionType::Solution => {
+                HashMap::from([("content".to_string(), payload.input.content.unwrap())])
+            }
+            EntityRevisionType::Video => HashMap::from([
+                ("content".to_string(), payload.input.content.unwrap()),
+                (
+                    "description".to_string(),
+                    payload.input.description.unwrap(),
+                ),
+                ("title".to_string(), payload.input.title.unwrap()),
+            ]),
```

Comment:
I'd move that part into api.serlo.org and just pass an object `fields` to the database layer. If you want, you could write some short validation code that checks whether the given `fields` contain all relevant data required for the specific entity. But even that you could skip, IMHO, since api.serlo.org has the validation via GraphQL.

Reply:
copy that.
Done in 90bb326 and in https://github.com/serlo/api.serlo.org/pull/525/commits/699e6e7b06fe0664020d447e30af3515b101c28f.
Please check if that implementation is what you suggested

Reply:
At the end of the day, since 90bb326 we are not using EntityRevisionType. Should I remove it from payload?

Reply:
I'm seeing now the disadvantage that the way I've implemented some fields are send in the mutation with a name, but retrieved with another (for example, CourseRevision sends a 'description' but its content it is fetched in a UuidQuery as 'content'; VideoRevision is sends a 'content', but in the UuidQuery it is a 'url').
I can try to improve that in a follow up PR.